funds-manager/funds-manager-server/src/handlers.rs

```diff
@@ -250,17 +250,16 @@ pub(crate) async fn swap_immediate_handler(
     let (augmented_quote, receipt, swap_gas_cost) =
         execution_client.swap_immediate_decaying(chain, params, wallet).await?;
 
-    let resp = SwapImmediateResponse {
+    // Compute swap costs and respond
+    let cost_data =
+        metrics_recorder.record_swap_cost(&receipt, &augmented_quote, swap_gas_cost).await?;
+    let execution_cost = cost_data.execution_cost_usdc;
+
+    Ok(warp::reply::json(&SwapImmediateResponse {
         quote: augmented_quote.quote.clone(),
         tx_hash: format!("{:#x}", receipt.transaction_hash),
-    };
-
-    // Record swap cost metrics
-    tokio::spawn(async move {
-        metrics_recorder.record_swap_cost(&receipt, &augmented_quote, swap_gas_cost).await;
-    });
-
-    Ok(warp::reply::json(&resp))
+        execution_cost,
+    }))
```

Comment:
IMO, if we fail to record swap costs, we shouldn't fail to send a successful response here. The bot server should still be able to observe a successful swap to proceed w/ its rebalancing logic in the even that execution costs aren't successfully added to the response